types/poi/index.d.ts

```diff
@@ -0,0 +1,223 @@
+
+export interface Opts {
+    type: string;
+    mode: Mode;
+    [k: string]: any;
+}
+
+/**
+ * https://poi.js.org/config.html
+ */
+export interface Config {
+    entry?: Config.Entry;
+    output?: Config.Output;
+    pages?: Config.Pages;
+    babel?: Config.Babel;
+    css?: Config.Css;
+    assets?: Config.Assets;
+    envs?: Config.Envs;
+    constants?: Config.Constants;
+    chainWebpack?: Config.ChainWebpack;
+    configureWebpack?: Config.ConfigureWebpack;
+    publicFolder?: Config.PublicFolder;
+    devServer?: Config.DevServer;
+    plugins?: Config.Plugins;
+}
+
+export namespace Config {
+    type Entry = WebpackConfig["entry"];
+
+    interface Output {
+        dir?: string;
+        clean?: boolean;
+        format?: "iife" | "cjs" | "umd";
+        moduleName?: string;
+        sourceMap?: boolean;
+        minimize?: boolean;
+        publicUrl?: string;
+        fileNames?: Output.FileNames;
+        target?:
+            | "web"
+            | "electron-renderer"
+            | "electron-main"
+            | "node"
+            | "node-webkit"
+            | "async-node"
+            | "webworker";
+        html?: Output.Html;
+    }
+    namespace Output {
+        interface FileNames {
+            js?: string;
+            css?: string;
+            font?: string;
+            image?: string;
+        }
+
+        interface Html {
+            title?: string;
+            filename?: string;
+            template?: string;
+            inject?: boolean;
+        }
+    }
+
+    interface Pages {
+        [pageName: string]:
+            | string
+            | Partial<HtmlWebpackPluginOptions> & {
+                  entry: string;
+                  chunks?: string[];
+              };
+    }
+
+    interface Babel {
+        jsx?: string;
+        transpileModules?: string | string[];
+    }
+
+    interface Css {
+        extract?: boolean;
+        sourceMap?: boolean;
+        loaderOptions?: Css.LoaderOptions;
+    }
+    namespace Css {
+        interface LoaderOptions {
+            css?: any;
+            sass?: any;
+            postcss?: any;
+            less?: any;
+            stylus?: any;
+        }
+    }
+
+    interface Assets {
+        inlineImageMaxSize?: number;
+    }
+
+    interface Envs {
+        [envName: string]: string;
+    }
+
+    interface Constants {
+        [constantName: string]: string;
+    }
+
+    interface ChainWebpack {
+        (config: WebpackChainConfig, opts: Opts): void;
+    }
+
+    type ConfigureWebpack =
+        | WebpackConfig
+        | ((config: WebpackConfig, opts: Opts) => void | WebpackConfig);
+
+    type PublicFolder = string | boolean;
+
+    interface DevServer {
+        host?: string;
+        port?: string | number;
+        hot?: boolean;
+        hotOnly?: string;
+        hotEntries?: string[];
+        historyApiFallback?: WebpackDevServerConfig["historyApiFallback"];
+        open?: boolean;
+        proxy?: WebpackDevServerConfig["proxy"];
```

Comment:
`proxy` can be a `string`

Reply:
Hi @egoist, I found proxy also support function, but I can not find the function signature both in [poi documention](https://poi.js.org/config.html#devserver-proxy) or [webpack dev server documention](https://webpack.js.org/configuration/dev-server/#devserverproxy)

Can you tell me where introduced the definition of this function?